subsys/storage/flash_map/flash_map_shell.c

```diff
@@ -22,21 +22,29 @@ extern const struct flash_area *flash_map;
 
 static void fa_cb(const struct flash_area *fa, void *user_data)
 {
-	struct shell *shell = user_data;
-
-	shell_print(sh, "%2d   0x%0*"PRIxPTR"   %-26s  0x%-10x 0x%-12x",
-		    (int)fa->fa_id, sizeof(uintptr_t) * 2, (uintptr_t)fa->fa_dev, fa->fa_dev->name,
-		    (uint32_t) fa->fa_off, fa->fa_size);
+	struct shell *sh = user_data;
+#if CONFIG_FLASH_MAP_LABELS
+	const char *fa_label = flash_area_label(fa);
+
+	if (fa_label == NULL) {
+		fa_label = "-";
```

Comment:
Change this to "<missing>" or something like this,, because now if you do not have any labels result is the same as when they have not been enabled. I am not a fun of '-' on the not enabled side, but have no better idea.